src/Controller/RefileRequestController.php

```diff
@@ -0,0 +1,120 @@
+     *     summary="Process a refile request",
+     *     tags={"recap"},
+     *     operationId="createRefileRequest",
+     *     consumes={"application/json"},
+     *     produces={"application/json"},
+     *     @SWG\Parameter(
+     *         name="NewRefileRequest",
+     *         in="body",
+     *         description="Request object based on the included data model",
+     *         required=true,
+     *         @SWG\Schema(ref="#/definitions/NewRefileRequest")
+     *     ),
+     *     @SWG\Response(
+     *         response=200,
+     *         description="Successful operation",
+     *         @SWG\Schema(ref="#/definitions/RefileRequestResponse")
+     *     ),
+     *     @SWG\Response(
+     *         response="401",
+     *         description="Unauthorized"
+     *     ),
+     *     @SWG\Response(
+     *         response="404",
+     *         description="Not found",
+     *         @SWG\Schema(ref="#/definitions/RefileRequestErrorResponse")
+     *     ),
+     *     @SWG\Response(
+     *         response="500",
+     *         description="Generic server error",
+     *         @SWG\Schema(ref="#/definitions/RefileRequestErrorResponse")
+     *     ),
+     *     security={
+     *         {
+     *             "api_auth": {"openid offline_access api write:hold_request readwrite:hold_request"}
+     *         }
+     *     }
+     * )
+     *
+     * @throws APIException
+     * @return Response
+     */
+    public function createRefileRequest()
+    {
+        try {
+            $data = $this->getRequest()->getParsedBody();
+
+            $refileRequest = new RefileRequest($data);
+
+            $refileRequest->create();
+
+            APILogger::addNotice('Beginning refile of item barcode ' . $data['itemBarcode']);
+
+            APILogger::addNotice('Getting item record');
+
+            $itemClient = new ItemClient();
+
+            $response = $itemClient->get('items?barcode=' . $data['itemBarcode']);
+
+            $item = json_decode($response->getBody(), true)['data'][0];
+
+            APILogger::addNotice('Received item record', $item);
+
+            APILogger::addNotice('Sending SIP2 call', [
+                'barcode' => $item['barcode'],
+                'locationCode' => $item['location']['code']
+            ]);
+
+            $sip2Client = new SIP2Client();
+
+            $checkinResponse = $sip2Client->getSip2Client()->msgCheckin(
+                $item['barcode'],
+                time(),
+                $item['location']['code']
+            );
+
+            $result = $sip2Client->getSip2Client()->parseCheckinResponse(
+                $sip2Client->getSip2Client()->get_message($checkinResponse)
+            );
+
+            APILogger::addNotice('Received SIP2 message', $result);
+
+            return $this->getResponse()->withJson(
+                new RefileRequestResponse($refileRequest)
+            );
+
+        } catch (\Exception $exception) {
```

Comment:
What types of `exceptions` can be caught by the sip2Client? we might need to translate various errors to RESTful errors with statusCodes?

Reply:
As far as I can tell, the SIP2 library we're using only returns simple error messages and debug information. There doesn't seem to be any exception handling. We may have to build this in ourselves at some point if necessary. For now, we're just trapping all success and all errors as the same.